oligo_designer_toolsuite/oligo_selection/_generate_oligosets.py

```diff
@@ -144,20 +149,21 @@ def _get_oligo_set_for_gene(
         oligos_scores.sort_values(ascending=self.ascending, inplace=True)
 
         # hard limit on the number of oligos
-        if len(oligos_scores) > self.max_oligos:
-            # select the best oligos
-            for oligo_id in oligos_scores.index[self.max_oligos :]:
-                del oligo_database.database[region_id][oligo_id]
-                oligos_scores.drop(oligo_id, inplace=True)
+        # if len(oligos_scores) > self.max_oligos:
```

Comment:
move to graph based

Reply:
It would have been too (unnecessarily) complicated to move it to the graph-based class. Adding the isinstance check for now

Reply:
Initialize max_oligos to None (infinity) by default